scipy/optimize/tests/test_optimize.py

```diff
@@ -157,7 +157,9 @@ def test_bfgs_nan(self):
         fprime = lambda x: np.ones_like(x)
         x0 = [np.nan]
         with np.errstate(over='ignore', invalid='ignore'):
-            x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', LineSearchWarning)
+                x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
```

Comment:
So the only valid filtering in this file is for direct calls to `line_search_wolfe2`.